Jenkins Build

ffalcn opened the pull request:
Initial addition of Jenkinsfile and helm files. I haven't been able to test this in Jenkins yet (having some permissions problems) but wanted to get the initial review started. I based this off of the v1.11.1 branch as it seemed the most up to date.

Commit:
Initial addition of Jenkinsfile and
helm files.

campbelldgunn commented on Jenkinsfile:
```diff
@@ -0,0 +1,47 @@
+      }
+
+      stage('deploy to production') {
+        standardRelease(config, version_tag) {}
```
@ffalcn Does it need to deploy to production? The fact that it is published to ACR as an official image means it is ready for use by production pipelines.

ffalcn replied:
I wasn't sure if it did or not, but left it in as it followed the example. I will remove it.

campbelldgunn commented on charts/k8s-kubectl/templates/deployment.yaml:
```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1beta2
```
@ffalcn I don't think you need the helm charts. As this is an image that is consumed by other pipelines to build and deploy from their respective pipelines?

ffalcn replied:
That makes sense, I will remove the charts.

campbelldgunn reviewed (changes requested):
I don't you need the chart files it is more about getting the image up to date that is consumed by other pipelines.

Commit:
removed helm charts
modified Jenkinsfile to remove deploy
to production

campbelldgunn reviewed (commented):
How did you validate this? I did not see it in the dev/sandbox of Jenkins.

ffalcn commented:
So this hasn't been validated yet, I don't yet have access to the Jenkins dev environment. Jake was working on getting the permissions for me yesterday, but there seems to have been a problem getting the changes through. I am checking with him now to see if he has any kind of update.

ffalcn commented:
In checking with Jake he hasn't had any luck getting the permissions sorted out and suggested we check with Bob when he gets back.

Commit:
modified to use the linuxTemplate
(misunderstood what the template meant)

Commit:
Found another example in the k8s-helm
repo, attempting some modifications based off of it.

Commit:
modify the nodeSelector

Commit:
Realized that it appears the template may
not have been necessary, so attempting
a simplified version of the Jenkinsfile.

Commit:
Based on errors it appears the node is required.

Commit:
using the linuxPodTemplate to wrap the node.

Commit:
removed check that doesn't work with config from parseJson method

Commit:
added alt_acct to avoid dockerfiles being pushed to null

ffalcn commented:
The branch and pull request builds now run successfully. I examined the logs from the build and didn't see any issues, is there anything else you'd like me to add to this?

campbelldgunn commented:
@ffalcn Can you make sure that anything in the docker image that is referencing the public is now referencing internal. I think there are few references in the docker file.

Commit:
Update label url to dfs repo

Commit:
modified Makefile and Readme to point to azure container repo
and dfs github repo.

campbelldgunn commented on README.md:
```diff
@@ -6,7 +6,7 @@
 [![](https://images.microbadger.com/badges/commit/campbelldgunn/k8s-kubectl.svg)](http://microbadger.com/images/campbelldgunn/k8s-kubectl "Get your own commit badge on microbadger.com")
```
We can remove the microbadger badges out as well.

Commit:
Removed microbadger images
adjusted readme format from linter suggestions
modified Maintainer field to dfsco email

Commit:
Changes as discussed in k8s-helm pull request:
removed hostname from app section
changed namespace to utilities

campbelldgunn reviewed (approved):
LGTM
